src/PluginBase.php

```diff
@@ -11,30 +11,31 @@
 use Composer\IO\IOInterface;
 use Composer\Plugin\PluginInterface;
 use PharIo\ComposerDistributor\Service\Installer;
-use SplFileInfo;
 
 abstract class PluginBase implements PluginInterface, EventSubscriberInterface
 {
+    /** @var \Composer\Composer */
     protected $composer;
 
+    /** @var \Composer\IO\IOInterface */
     protected $io;
```

Comment:
Do these have to be `protected` for some reason? If not, make the them `private`?

Reply:
All child classes of `PluginBase` need to access `$composer` and `$io` so I think it should stay protected.

Reply:
We do control these classes, right?

So then the answer is a clear: NO.
Properties are always private. If we need to access them in child classes, then we have to add a getter. Otherwise we'd break the `o` of solid.

Reply:
You are right. I changed the properties to private and added a protected getter to access them.

Reply:
Thanks :-) :+1: